File: modulo2/shopapp/app/src/main/java/com/shopapp/domain/repository/CategoryRepository.kt
// domain/repository/CategoryRepository.kt
package com.shopapp.domain.repository

import com.shopapp.domain.model.Category
import com.shopapp.domain.model.CategoryPayload

interface CategoryRepository {
    suspend fun getCategories(): Result<List<Category>>
    suspend fun getCategory(id: Int): Result<Category>
    suspend fun createCategory(payload: CategoryPayload): Result<Category>
    suspend fun updateCategory(id: Int, payload: CategoryPayload): Result<Category>
    suspend fun deleteCategory(id: Int): Result<Unit>
    suspend fun getStats(): Result<Map<String, Any>>
}

File: modulo4/flutter_shop_app/lib/data/repository/category_repository.dart
// lib/domain/repository/category_repository.dart


import 'package:flutter_shop_app/domain/model/category.dart';

abstract class CategoryRepository {
  Future<List<Category>> getCategories();
  Future<Category>       getCategory(int id);
  Future<Category>       createCategory(Map<String, dynamic> payload);
  Future<Category>       updateCategory(int id, Map<String, dynamic> payload);
  Future<void>           deleteCategory(int id);
  Future<Map<String, dynamic>> getStats();
}